recipes/libusb/all/conanfile.py

```diff
@@ -0,0 +1,133 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, MSBuild, tools
+import os
+
+class LibUSBConan(ConanFile):
+    name = "libusb"
+    description = "A cross-platform library to access USB devices"
+    license = "LGPL-2.1"
+    homepage = "https://github.com/libusb/libusb"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("conan", "libusb", "usb", "device")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "enable_udev": [True, False], "fPIC": [True, False]}
+    default_options = {'shared': False, 'enable_udev': True, 'fPIC': True}
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+
+    @property
+    def _is_msvc(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "Visual Studio"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = self.name + "-" + self.version
+        os.rename(extracted_folder, self._source_subfolder)
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.settings.os != "Linux":
+            del self.options.enable_udev
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def system_requirements(self):
+        if self.settings.os == "Linux":
+            if self.options.enable_udev:
+                package_tool = tools.SystemPackageTool(conanfile=self)
+                libudev_name = ""
+                os_info = tools.OSInfo()
+                if os_info.with_apt:
+                    libudev_name = "libudev-dev"
+                elif os_info.with_yum:
+                    libudev_name = "libudev-devel"
+                elif os_info.with_zypper:
+                    libudev_name = "libudev-devel"
+                elif os_info.with_pacman:
+                    libudev_name = "libsystemd systemd"
+                else:
+                    self.output.warn("Could not install libudev: Undefined package name for current platform.")
+                    return
+                package_tool.install(packages=libudev_name, update=True)
```

Comment:
Can we change this to a check install of an install please?

Reply:
systemd is too low-level, but I'm still against system_requirements installations in recipes

Reply:
Agreed, I need to fix it on Windows. I only tested on my Linux env, but let me update is first

Reply:
@Croydon SystemPackageTool.install() already checks if it's installed: https://github.com/conan-io/conan/blob/develop/conans/client/tools/system_pm.py#L120

Reply:
Sorry, I wrote my comment above pretty confusing

I meant, can we write this in a way that Conan is **_only_** checking if these system packages are installed **_without_** trying to automatically install it?

Reply:
Yes, we can. The method is protected, but it's possible calling. What's is your idea, only a warning when not installed ?

Reply:
Warning is probably the best.

It might be a hard dependency, but since we can never cover all distributions, their system package manager and all of the different system package names, it could mean that the package IS installed and we simply can't detect it right. So a warning should be fine

Reply:
sounds plausible. I'll apply it

Reply:
done: https://github.com/conan-io/conan-center-index/pull/430/commits/80cf33a01d1b987be660a8540195b3494656685a

Reply:
@Croydon now the CI build is kaputt, because libudev is required by default.

Reply:
Any chance we could install this be default?

Reply:
well, we would need to change our docker images, but installing libudev only because libusb package sounds wrong. WDYT @danimtb ?

Reply:
I'm seeing it this way: 

Either we think it is too low level, then we should pre-install it, because we definitely shouldn't install system requirements within recipes anymore.

Or it isn't too low level and then we should package it via Conan

🤔

Reply:
> we definitely shouldn't install system requirements within recipes anymore.

It's not totally true. I discussed it with Conan team some time ago. Some packages like libudev we can do nothing about. Also if you don't want to install it, you can set `CONAN_SYSREQUIRES_MODE=disabled`

I just reverted libudev installation, otherwise, this PR won't walk forward.